src/sentry/api/bases/organization_events.py

```diff
@@ -194,19 +216,26 @@ def handle_data(self, request, organization, project_ids, results):
         return results
 
     def get_event_stats_data(
-        self, request, organization, get_event_stats, top_events=False, query_column="count()"
+        self,
+        request,
+        organization,
+        get_event_stats,
+        top_events=False,
+        query_column="count()",
+        params=None,
     ):
         with self.handle_query_errors():
             with sentry_sdk.start_span(
                 op="discover.endpoint", description="base.stats_query_creation"
             ):
                 columns = request.GET.getlist("yAxis", [query_column])
                 query = request.GET.get("query")
-                try:
-                    params = self.get_filter_params(request, organization)
-                except NoProjects:
-                    return {"data": []}
-                params = self.quantize_date_params(request, params)
+                if params is None:
+                    try:
+                        params = self.get_snuba_params(request, organization)
+                    except NoProjects:
+                        return {"data": []}
```

Comment:
Are we able to update all the callers so that this code path isn't required?

Reply:
Unfortunately not, some of the endpoints act differently when there are no projects. Eg. related-issues returns an empty list, but here we return a dictionary instead.

Reply:
That's unfortunate.